java/acceptancetest/step_definitions/jmri/web/HomeServletAcceptanceSteps.java

```diff
@@ -0,0 +1,51 @@
+package jmri.web;
+
+import cucumber.api.java8.En;
+import cucumber.api.PendingException;
+import org.junit.Assert;
+import org.openqa.selenium.support.events.EventFiringWebDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxBinary;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import java.awt.GraphicsEnvironment;
+import jmri.web.server.WebServer;
+
+/**
+ * Cucumber step defintions for Home Servlet Acceptance tests.
+ *
+ * @author  Paul Bender Copyright (C) 2017
+ */
+public class HomeServletAcceptanceSteps implements En {
+     
+   private EventFiringWebDriver webDriver;
+   
+   String[] tags = {"@webtest"};
+   
+   public HomeServletAcceptanceSteps() {
+
+      Before(tags,()->{
+         if(GraphicsEnvironment.isHeadless()) {
+            FirefoxBinary firefoxBinary = new FirefoxBinary();
+            firefoxBinary.addCommandLineOptions("--headless");
+            FirefoxOptions firefoxOptions = new FirefoxOptions();
+            firefoxOptions.setBinary(firefoxBinary);
+            webDriver = new EventFiringWebDriver(new FirefoxDriver(firefoxOptions));
+         } else {
+            webDriver = new EventFiringWebDriver(new FirefoxDriver());
+         }
+      });
+
+      When("^I ask for the /index\\.html$", () -> {
```

Comment:
Do note that the home servlet does not return a web page on a request for /index.html, and this test is not requesting /index.html, so this comment indicates this test is for the wrong resource, or is wrong itself.

Reply:
I'll fix that.

BTW, that's actually not a comment.  That's a behavior statement.  Cucumber is really about behavior driven testing.

This is (eventually) going to be a parameterized test that describes things we ask for and what is returned, so we really care about the result, not the specific object providing the result.

Reply:
The result of requesting /index.html is to provide an HTTP 302 response without a body, which is different than the HTTP 200 response with a body you get from requesting /. I don't know what you are expecting since different clients can treat that 302 differently.

Reply:
Since we are doing this after the fact, we're going to end up documenting the behavior with the tests.

I'll need to investigate and see if we get the status codes from the headers or can just interact with the windows.

Should we be responding with either a 303 or a 307 instead of a 302?